### PR TITLE
[Site] More Functional Code, 2x Complex Live Components Examples, few fixes

### PR DESCRIPTION
**Pull request by weaverryan**
| Q             | A
| ------------- | ---
| Bug fix?      | yes (with respect to Twig errors)
| New feature?  | no
| Tickets       | None
| License       | MIT

Hi!

Prepping ux.symfony.com for 2.8 - very excited about this! Join me on a tour:

## 1) Invoice Creator Live Component Demo

https://user-images.githubusercontent.com/121003/232886502-7f8da8f5-fe8a-4251-8e92-c2b56895b1ee.mp4

Parent component + collection of child components communicating by emitting events!

## 2) Product form + modal to create a Category

https://user-images.githubusercontent.com/121003/232888282-ab12c4a1-8153-4fe5-babf-cbf4a313c47b.mp4

Opens a Bootstrap modal with a child component inside - and closes that modal from a `LiveAction`.

### 3) Much nicer code blocks

Before:

Screenshot 2023-04-18 at 3 45 34 PM

After:

Screenshot 2023-04-18 at 3 45 41 PM

* "Copy" button
* Link to view the file on GitHub
* use statements are collapsed (click to show them)
* "Expand code" to show longer code blocks

### 4) Explanations on longer examples

Some examples deserve pointing out a few of the more complex parts. Some of the live component demos now do this:

Screenshot 2023-04-18 at 3 48 42 PM

### 5) Using a LOT of new features

[Smart rendering](https://symfony.com/bundles/ux-live-component/current/index.html#the-smart-re-render-algorithm), [emitting events](https://symfony.com/bundles/ux-live-component/current/index.html#communication-between-components-emitting-events), [browser events](https://symfony.com/bundles/ux-live-component/current/index.html#dispatching-browser-javascript-events), etc, etc - MANY new features are being used on the site now. 

We're also using (in som
e places) the new `<twig:Component>` syntax from #771. Even though editor support (e.g. for auto-completion) doesn't exist yet, I LOVE this. For example, to render a code block:

```html
<twig:CodeBlock filename="src/Twig/SearchPackages.php" />

{# or more complex - logic for extracting" a specific Twig block lives in the CodeBlock.php component #}
<twig:CodeBlock
    filename="templates/main/homepage.html.twig"
    targetTwigBlock="markdown_example"
    :showTwigExtends="false"
    :stripExcessHtml="true"
/>
```

Thanks everyone for the work, review & testing on a lot of this new stuff. Using the new features on the site has been BLAST.

### CODE REVIEW

File: src/TwigComponent/src/Twig/ComponentExtension.php [hunk 1]
```diff
@@ -53,11 +54,27 @@ public function getTokenParsers(): array
...
+            $this->throwRuntimeError($name, $e);
+        }
+    }
+
+    private function throwRuntimeError(string $name, \Throwable $e): void
+    {
+        if (!($e instanceof \Exception)) {
+            $e = new \Exception($e->getMessage(), $e->getCode(), $e->getPrevious());
+        }
+        throw new RuntimeError(sprintf('Error rendering "%s" component: %s', $name, $e->getMessage()), previous: $e);
```
Comment (weaverryan) [inline src/TwigComponent/src/Twig/ComponentExtension.php:78]: I snuck this feature in. We may need some trial and error. But throwing a `RuntimeError` allows Twig to show the actual template + line number where the error occurred. The nested exception will be the one from your component (e.g. maybe you are passing `null` to a property in your component that only accepts `string`).

File: src/TwigComponent/src/Twig/TwigPreLexer.php [hunk 2]
```diff
@@ -75,7 +77,7 @@ public function preLexComponents(string $input): string
                 $lastComponentName = $lastComponent['name'];
 
                 if ($closingComponentName !== $lastComponentName) {
-                    throw new \RuntimeException("Expected closing tag '</twig:{$lastComponentName}>' but found '</twig:{$closingComponentName}>' at line {$this->line}");
+                    throw new SyntaxError("Expected closing tag '</twig:{$lastComponentName}>' but found '</twig:{$closingComponentName}>'", $this->line);
```
Comment (weaverryan) [inline src/TwigComponent/src/Twig/TwigPreLexer.php:84]: This is just a correction: allows the lexing error to show the template + line number in the exception.

**Comment by jmsche** [comment]
> use statements are collapsed (click to show them)

Maybe indicate somehow that it is clickable? (more descriptive text, icon...)

**Comment by weaverryan** [comment]
> use statements are collapsed (click to show them)

> Maybe indicate somehow that it is clickable? (more descriptive text, icon...)

Done! We can iterate on it if it's still not obvious

**Pull request merged**